### PR TITLE
Forge on

### LINKED ISSUE #651: Idea for missing fields and topologies
emit a warning on rank 0. Then just pass empty collections down the line.

### PR DESCRIPTION
**Pull request by mclarsen**
resolves #651

### CODE REVIEW

File: src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp [hunk 1]
```diff
@@ -208,9 +208,14 @@ VTKHMarchingCubes::execute()
     std::shared_ptr<VTKHCollection> collection = data_object->as_vtkh_collection();
 
     std::string field_name = params()["field"].as_string();
+
     if(!collection->has_field(field_name))
     {
-      detail::field_error(field_name, this->name(), collection);
+      bool throw_error = false;
```
Comment (mclarsen) [inline src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp:214]: This is the basic idea of skipping

File: src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp [hunk 2]
```diff
@@ -208,9 +208,14 @@ VTKHMarchingCubes::execute()
...
+      detail::field_error(field_name, this->name(), collection, throw_error);
+      // this creates a data object with an invalid soource
+      set_output<DataObject>(new DataObject());
+      return;
```
Comment (mclarsen) [inline src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp:224]: This is the basic idea of failing. Then we just have some special handling in the rendering.

File: src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp [hunk 3]
```diff
@@ -205,6 +205,12 @@ VTKHMarchingCubes::execute()
     }
 
     DataObject *data_object = input<DataObject>(0);
+    if(!data_object->is_valid)
+    {
+      set_output<DataObject>(data_object);
+      return;
+    }
+
```
Comment (mclarsen) [inline src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp]: Each filter would just do this check.

**Review by cyrush — APPROVED** [review]
Looks good, this is as nice way to manage this!

**Pull request merged**